curp/src/server/curp_node.rs

```diff
@@ -173,42 +176,109 @@ impl<C: 'static + Command> CurpNode<C> {
 
         let mut ticker = tokio::time::interval(heartbeat_interval);
         ticker.set_missed_tick_behavior(MissedTickBehavior::Delay);
-        #[allow(clippy::integer_arithmetic, unused_attributes)] // tokio internal triggered
         loop {
             let _now = ticker.tick().await;
-            let action = curp.tick();
-            match action {
-                TickAction::Heartbeat(hbs) => {
-                    Self::bcast_heartbeats(Arc::clone(&curp), &connects, hbs).await;
-                }
-                TickAction::Votes(votes) => {
-                    Self::bcast_votes(Arc::clone(&curp), &connects, votes).await;
-                }
-                TickAction::Nothing => {}
+            if let Some(vote) = curp.tick_election() {
+                Self::bcast_vote(curp.as_ref(), &connects, vote).await;
             }
         }
     }
 
-    /// Background leader calibrate followers
-    async fn calibrate_task(
+    /// Sync task is responsible for syncing a follower
+    async fn sync_task(
         curp: Arc<RawCurp<C>>,
-        connects: HashMap<ServerId, Arc<impl ConnectApi>>,
-        mut calibrate_rx: mpsc::UnboundedReceiver<ServerId>,
+        connect: Arc<impl ConnectApi>,
+        sync_event: Arc<Event>,
+    ) {
+        let leader_event = curp.leader_event();
+        loop {
+            if !curp.is_leader() {
+                leader_event.listen().await;
+            }
+            Self::_sync_task(
+                Arc::clone(&curp),
+                Arc::clone(&connect),
+                Arc::clone(&sync_event),
+            )
+            .await;
+        }
+    }
+
+    /// real sync task, will stop if self is no long the leader
+    async fn _sync_task(
```

Comment:
The leading `_` has has a special meaning in rust, so I think the naming here is not very good, I suggest use `inner_sync_task`

Reply:
Changed to sync_follower